### PR TITLE
Update contributing doc

### PR DESCRIPTION
**Pull request by rwstauner**
Add a few things I learned from #179 to `CONTRIBUTING.md`

-----------------

Before submitting the PR make sure the following are checked:

* [x] The PR relates to *only* one subject with a clear title and description in grammatically correct, complete sentences.
* [x] Wrote [good commit messages][1].
* [x] Commit message starts with `[Fix #issue-number]` (if the related issue exists).
* [x] Feature branch is up-to-date with `master` (if not - rebase it).
* [x] Squashed related commits together.
* [x] Added tests.
* [x] Ran `bundle exec rake default`. It executes all tests and runs RuboCop on its own code.
* [x] Added an entry (file) to the [changelog folder](https://github.com/rubocop/rubocop-minitest/blob/master/changelog/) named `{change_type}_{change_description}.md` if the new code introduces user-observable changes. See [changelog entry format](https://github.com/rubocop/rubocop/blob/master/CONTRIBUTING.md#changelog-entry-format) for details.

[1]: https://chris.beams.io/posts/git-commit/

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -33,6 +33,7 @@ $ rubocop -V
 * If your change has a corresponding open GitHub issue, prefix the commit message with `[Fix #github-issue-number]`.
 * Make sure to add tests for it. This is important so I don't break it
   in a future version unintentionally.
+* If you are adding a new cop add an entry for it in `config/default.yml`.  Use `VersionAdded: '<<next>>'`.
```
Comment (koic) [inline CONTRIBUTING.md:36]: I think it's probably better to explanation rake task `new_cop` instead.

```suggestion
* If you are adding a new cop [use the bundled rake task `new_cop` to generate a cop template][7].
```

If so, please update the entry and add the following link at the end.

```markdown
[7]: https://docs.rubocop.org/rubocop/development.html
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -59,7 +60,7 @@ Here are a few examples:
 * If the change has a related GitHub issue (e.g. a bug fix for a reported issue), put a link to the issue as `[#123](https://github.com/rubocop/rubocop-minitest/issues/123): `.
 * Describe the brief of the change. The sentence should end with a punctuation.
 * At the end of the entry, add an implicit link to your GitHub user page as `([@username][])`.
-* Alternatively, you may modify the CHANGELOG file directly, but this may result in conflicts later on. Also, if this is your first contribution to RuboCop Minitest project, add a link definition for the implicit link to the bottom of the changelog as `[@username]: https://github.com/username`.
+* Alternatively, you may modify the CHANGELOG file directly, but this may result in conflicts later on.
```
Comment (koic) [inline CONTRIBUTING.md:63]: This entire entry line seems unnecessary.

```suggestion
```

File: CONTRIBUTING.md (hunk 1 above)
Comment (rwstauner) [reply CONTRIBUTING.md]: indeed, that's what i should have started with 😅

**Comment by koic** [comment]
Thanks!

**Pull request merged**